### PR TITLE
Adv/update serialization docs

### PR DESCRIPTION
**Pull request by antonydellavecchia**
Some updates to serialization documentation.

Some links from the general documentation have been added, including a link to the developer document.

Details about external implementation has been added, and an example on how polynomials are encoded.

Related to https://github.com/oscar-system/Oscar.jl/issues/3761

### CODE REVIEW

**Comment by micjoswig** [comment]
I made several small changes. Fine with me for now.

**Review by micjoswig — APPROVED** [review]
@benlorenz : I did not merge the current master into this branch, to avoid possible issues with backporting.  That's also why I do not merge this branch into the master.  Please take care of this.

File: docs/src/DeveloperDocumentation/serialization.md [hunk 1]
```diff
@@ -31,16 +32,69 @@ as follows:
...
+The content is "42", it represents a `Base.Int`, according to the `_type` field.
 
 
 ## Implementation
-All files for serialization can be found in the folder `src/Serialization`. The
-naming conventions of the files there follows the overall structure of OSCAR,
-i.e. the file `src/Serialization/PolyhedralGeometry.jl` contains functions for
+To list and describe all implementations and encodings of all types in OSCAR
+is not a possible feat due to the arbitrarily deep and nested type structures
+used available in OSCAR, we point any developer looking to understand the encodings
```
Comment (joschmitt) [inline docs/src/DeveloperDocumentation/serialization.md]:
```suggestion
available in OSCAR, we point any developer looking to understand the encodings
```

File: docs/src/DeveloperDocumentation/serialization.md [hunk 2]
```diff
@@ -31,16 +32,69 @@ as follows:
...
+of certain types to the OSCAR source code. All files for serialization can be
+found in the folder `src/Serialization`. The  conventions of the files there
```
Comment (joschmitt) [inline docs/src/DeveloperDocumentation/serialization.md]:
```suggestion
found in the folder `src/Serialization`. The convention of the files there
```

File: docs/src/DeveloperDocumentation/serialization.md [hunk 3]
```diff
@@ -31,16 +32,69 @@ as follows:
...
+follows the overall structure of OSCAR, i.e. the file
+`src/Serialization/PolyhedralGeometry.jl` contains functions for
 serializing objects of the polyhedral geometry section.
 
+We include a basic example of the encoding for a `QQPolyRingElem` so one can get a taste
+before delving into the source code. Here we store the polynomial `x^3 + 2x + 1//2`.
+The encoding for polynomials is to store a list of tuples, where each entry in the
+list represents a term of the polynomial. Where the first entry of the tuple is
```
Comment (joschmitt) [inline docs/src/DeveloperDocumentation/serialization.md]:
```suggestion
list represents a term of the polynomial and where the first entry of the tuple is
```

File: docs/src/DeveloperDocumentation/serialization.md [hunk 4]
```diff
@@ -31,16 +32,69 @@ as follows:
...
+the exponent and the second entry is the coefficient. Here we serialize a univariate
+polynomial so the first entries are always integers, in general this may be an array
+of integers. The coefficients here are elements of `QQ` however in general
+the coefficients themselves may be described as polynomials of the generators
+of some field extension, i.e. the second entry may again be a list of tuples and so on.
+The nested structure of the coefficient will depending on the description of the field
```
Comment (joschmitt) [inline docs/src/DeveloperDocumentation/serialization.md]:
```suggestion
The nested structure of the coefficient will depend on the description of the field
```

File: docs/src/DeveloperDocumentation/serialization.md [hunk 5]
```diff
@@ -298,3 +352,34 @@ The ramifications of making mathematical data FAIR are manifold.
...
+the schema. In the same way we cannot guarantee that any files created externally
+are valid in terms of the mathematics either, these will not lead to a parse error
+but instead will be handle as though the incorrect input has been passed to one
+of the Oscar functions.
+
+External implementations should not be expected to read or write all possible Oscar types.
+It is perfectly valid for external implementations to throw parse errors when a certain
+file format is unexpected. For example Oscar will parse a `QQFieldElem` that has data value
+"0 0 7 // - 1 0" as `-7//10`, even though this is not how it is serialized. We feel
+we should not restrict users when deserialize to formats that may have issues deserializing
```
Comment (joschmitt) [inline docs/src/DeveloperDocumentation/serialization.md]:
```suggestion
we should not restrict users when deserializing to formats that may have issues deserializing
```

Comment (antonydellavecchia) [reply docs/src/DeveloperDocumentation/serialization.md]: thanks!

**Pull request merged**